app/src/mainwindow2.cpp

```diff
@@ -1213,6 +1215,7 @@ void MainWindow2::makeConnections(Editor* pEditor, TimeLine* pTimeline)
     connect(pEditor, &Editor::updateTimeLine, pTimeline, &TimeLine::updateUI);
 
     connect(pEditor->layers(), &LayerManager::currentLayerChanged, mToolOptions, &ToolOptionWidget::updateUI);
+
```

Comment:
Removed line.
About the commit messages: I mostly do coding at home. Sometimes, If I'm going somewhere where I have some time to code, I stop in the middle of something, make a "work in progress" commit, and push it, so it is available through a pull, when I reach my destination. I have read the guidelines.
Now I am at it: Thanks a lot for your fine and gentle suggestions. I agree on all issues, and they all come down to my shortcomings as a programmer. I learned programming at a coding academy from 1999-2001. We worked with Delphi (Pascal) and JAVA. I've never been working with programming on a professional level, and I've never learned C. My first encounter with C++, was when I discovered Qt, some 10-12 years ago. I'm self taught. I didn't knew that you could have getter functions in the header file, I have never really used format codes (%), and I don't know what assert is. 
It is up-hill, as we say here, but if you and the team will be patient, I'm eager to learn more efficient methods than I'm used to, and, at the end of the day, I think I've proved that I can provide code that works.

Reply:
Don't worry David, take your time, since you're willing to learn and are helping the project in such ways, I'm sure everyone is more than happy to assist you when needed, be it polishing code or providing resources for learning on the off-side :slightly_smiling_face: 

As for what an assert(ion) is, as far as I understand, it's basically a statement where the boolean condition would always returns true at that point in time when the code is executed. So a false assertion as was suggested would result always in a "NOT true" return value, however I'm not sure if @J5lx actually wants you to use the assert function that comes with C++ as explained here: https://stackoverflow.com/questions/1571340/what-is-the-assert-function so we 'd have to wait for him to reply to your specific queries.

Reply:
@davidlamhauge Don’t sweat that stuff. I’m self-taught myself and I have been where you are now. Actually a lot of what I commented on is stuff that comes with experience, not stuff that you would learn in a course or in a book. I don’t think I’ve seen anything about assertions or the ordering of the if branches that I commented on somewhere else in any sort of course or book, ever (just to give some examples). Likewise, git and especially the conventions surrounding it are even less often touched upon. I make an effort to give other people hints where I see an opportunity, but I don’t think any less of others if they make such mistakes, because I know I once made them myself, and sometimes I do even now. When my comments on your code sound terse or rough, that’s because I’ve got more than a thousand lines to look at and I don’t want to spend lots of time typing, not because I’m super annoyed or anything.

As for assertions, they are basically a way to check for conditions that must always be true when the code works as intended. If it doesn’t, they cause an error to be generated and the program will abort (i.e., it “crashes”), so that the error cannot go unnoticed. As a rule of thumb, assertions are generally used to check for possible mistakes that would be located in the codebase you are working on, i.e. code which you yourself have control over, while errors originating elsewhere should cause an exception. That’s because in theory you can make sure that your own codebase is 100% bug-free and that all assertions are always true. And for the same reason, assertions are generally disabled in release versions, so that the condition never gets executed or checked – after all you made sure during development that there are no bugs in your code (I mean, you did, right?). Therefore, don’t be afraid to use computationally expensive checks in assertions when you see them fit – there won’t be any slowdown in releases. Lastly, false assertions are assertions which always evaluate to false and always fail. They are used in places which should never, ever be reached when the code is executed.

Regarding what Jose said, in Pencil2D code you’d generally want to use Q_ASSERT for assertions since we’re using Qt (i.e. `Q_ASSERT(false);` for a false assertion).